dataeng/jobs/analytics/AnswerDistribution.groovy

```diff
@@ -33,10 +32,15 @@ class AnswerDistribution {
                         string('OPSGENIE_HEARTBEAT_CONFIG_KEY', 'opsgenie_heartbeat_config_key')
                     }
                 }
+                publishers {
+                    postBuildTask {
+                        task('Finished', 'if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && [ -n "$OPSGENIE_HEARTBEAT_CONFIG_KEY" ] then curl -X POST "https://api.opsgenie.com/v2/heartbeats/$OPSGENIE_HEARTBEAT_NAME/disable" --header "Authorization: GenieKey $OPSGENIE_HEARTBEAT_CONFIG_KEY" fi', true)
```

Comment:
I don't like that this code is duplicated in each job file. Is there a way to include a separate shell file here? I couldn't find one.

Reply:
or, preferrably use more functional semantics by defining it with parameters: `opsgenie_heartbeat_publisher(name, config_key)`, then define the function:
```groovy
    public static def opsgenie_heartbeat_publisher = { name, config_key ->
        return {
            postBuildTask {
                ...
            }
        }
    }
```